### PR TITLE
Localized artifacts now contain Authenticode signed assemblies

### PR DESCRIPTION
**Pull request by donnie-msft**
# Bug

Fixes: 

## Description

Documentation update for https://github.com/NuGet/NuGet.Client/pull/6170

## PR Checklist

- [ ] Meaningful title, helpful description and a linked NuGet/Home issue
- [ ] Added tests
- [ ] Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.

### CODE REVIEW

**Review by jeffkl — DISMISSED** [review]

**Review by jeffkl — DISMISSED** [review]

**Review by jgonz120 — DISMISSED** [review]

**Review by Nigusu-Allehu — DISMISSED** [review]

File: docs/localizability.md [hunk 1]
```diff
@@ -24,9 +24,11 @@ Unfortunately, NuGet.Client does not use OneLocBuild, and any resource comments
 Although the steps below are for the dev branch, the process is similar for release branches.
 
 1. A PR merges into the dev branch.
-1. A NuGet.Client build generates a `localizationArtifacts` artifact.  This artifact contains newly built (but unsigned) assemblies and .lcg files.  The .lcg files contain information extracted from .resx files and are the primary input for later localization.
+1. A NuGet.Client build generates a `localizationArtifacts` artifact.  This artifact contains newly built Authenticode-signed assemblies [(see PR#6170)][12]) containing English resources and .lcg files.
```
Comment (zivkan) [inline docs/localizability.md]:
```suggestion
1. A NuGet.Client build generates a `localizationArtifacts` artifact.  This artifact contains newly built assemblies containing English resources and .lcg files.
```

If someone needs to investigate the loc process in a year or more from now, I'm not sure why the PR would be interesting to call out specifically.  In fact, the fact that it's signed or not also doesn't seem important to me.

The loc team's tools don't have a requirement about signed or unsigned assemblies.

File: docs/localizability.md [hunk 2]
```diff
@@ -24,9 +24,11 @@ Unfortunately, NuGet.Client does not use OneLocBuild, and any resource comments
...
+The .lcg files contain information extracted from .resx files and are the primary input for later localization.
 1. The localization team grabs the `localizationArtifacts` artifact from a NuGet.Client build.
-1. The localization team localizes strings and merges a ["LEGO" PR][5] with localized strings (in .lcl files).
+1. The localization team localizes strings.
+The signed assemblies are used to validate the exported .lcl won't break the build before the team merges a ["LEGO" PR][5] with localized strings (in .lcl files).
```
Comment (zivkan) [inline docs/localizability.md]:
```suggestion
The assemblies are used to validate the exported .lcl won't break the build before the team merges a ["LEGO" PR][5] with localized strings (in .lcl files).
```

File: docs/localizability.md [hunk 3]
```diff
@@ -149,4 +151,5 @@ private void DoSomething(ILogger logger)
 [8]: https://github.com/NuGet/NuGet.Build.Localization/blob/931c5f21742fff61c7f53a19039b89dddc7a01cd/localize/comments/15/NuGet.Packaging.dll.lci#L58-L66
 [9]: https://github.com/dotnet/sign/blob/ef0e6b3ef8281dff1d62cea34445bd88fc3e6714/src/Sign.Core/Resources.resx#L131-L134
 [10]: https://aka.ms/allaboutloc
-[11]: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/20709/MicroBuild-Localization
\ No newline at end of file
+[11]: https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/20709/MicroBuild-Localization
+[12]: https://github.com/NuGet/NuGet.Client/pull/6170
```
Comment (zivkan) [inline docs/localizability.md]: Similar to another comment, it's not clear to me why someone reading these docs would be more interested in this pull request than any other pull request that modified `build/loc.proj` or the build yaml.

File: docs/localizability.md (hunk 1 above)
Comment (donnie-msft) [reply docs/localizability.md]: > The loc team's tools don't have a requirement about signed or unsigned assemblies.

This documentation is for the NuGet Client team, not the loc team.

> In fact, the fact that it's signed or not also doesn't seem important to me.

The original document indicated that these were unsigned assemblies. This is an update reflecting the new state of these assemblies.
This change is part of the SFI work to ensure we sign assemblies. If localized assemblies are not signed, we are no longer meeting SFI requirements.

File: docs/localizability.md (hunk 2 above)
Comment (donnie-msft) [reply docs/localizability.md]: Consolidating to thread: https://github.com/NuGet/NuGet.Client/pull/6172#discussion_r1866718472

File: docs/localizability.md (hunk 3 above)
Comment (donnie-msft) [reply docs/localizability.md]: This pull request contains an explanation of why it's important to sign the localized assemblies, as well as how it's being done. It's not accidental that they're signed, it is intentional, so the documentation is referencing that PR to point to the purpose of that behavior.

**Review by zivkan — APPROVED** [review]

**Review by Nigusu-Allehu — APPROVED** [review]

**Pull request merged**